app/challenge.ts

```diff
@@ -0,0 +1,96 @@
+import { web3, workspace } from '@project-serum/anchor';
+import { TypeDef } from '@project-serum/anchor/dist/cjs/program/namespace/types';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import { expect } from 'chai';
+
+// full ts class meant to mimic the challenge PDA
+// pub initiator: Pubkey,
+// pub initiator_tokens_mint: Pubkey,
+// pub initiator_tokens_vault: Pubkey,
+// pub initiator_wager_token_amount: u64,
+// pub acceptor: Pubkey,
+// pub acceptor_tokens_mint: Pubkey,
+// pub acceptor_tokens_vault: Pubkey,
+// pub acceptor_wager_token_amount: u64,
+// pub acceptor_wager_approved: bool,
+// pub requester: Pubkey,
+// pub bump: u8,
+
+export class Challenge {
+    // part of initialization
+    seed: string = "challenge";
+    programId: PublicKey = workspace.CrossPile.programId;
```

Comment:
This won't work if implemented by client side code. Can we pass in the programId? If we release a client library we can then export it as a constant as part of the NPM package like: https://github.com/demox-labs/solrand/blob/main/index.js#L4